### PR TITLE
Lots of cleanups and restructuring to pave way for Django 1.6/1.7

### PR DESCRIPTION
**Pull request by mithro**

### CODE REVIEW

File: website/manage.py [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-    sys.stderr.write("Error: Can't find the file 'settings.py' in the directory containing %r. It appears you've customized things.\nYou'll have to run django-admin.py, passing it your settings module.\n" % __file__)
-    sys.exit(1)
-
-import settings
+import os
+import sys
 
 if __name__ == "__main__":
-    execute_manager(settings)
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "website.settings")
```
Comment (aps-sids) [inline website/manage.py]: This fails since there is no `__init__.py` in `website/`.

File: website/website/settings.py [hunk 2]
```diff
@@ -130,8 +124,6 @@
 )
 
 INSTALLED_APPS = OUR_APPS + (
-    'django_extensions',
```
Comment (aps-sids) [inline website/website/settings.py]: `django_extensions` is a requirement for `switter`.

**Comment by aps-sids** [comment]
`south` is no longer needed in Django>=1.7

File: website/website/settings.py [hunk 3]
```diff
@@ -50,23 +50,18 @@
...
 # Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
-MEDIA_URL = ''
+MEDIA_URL = '/media/'
 
 # Absolute path to the directory static files should be collected to.
 # Don't put anything in this directory yourself; store your static files
 # in apps' "static/" subdirectories and in STATICFILES_DIRS.
 # Example: "/home/media/media.lawrence.com/static/"
-STATIC_ROOT = '/tmp/timvideos-static'
+STATIC_ROOT = ''
```
Comment (aps-sids) [inline website/website/settings.py]: This is required by `staticfiles` app. Either add this or remove that.

File: website/requirements.txt [hunk 4]
```diff
@@ -2,18 +2,12 @@
...
+django>=1.6
 BeautifulSoup4
 django-live-server
-distribute>=0.6.14
-django-extensions
-ez-setup
 gunicorn
-ordereddict
 pylint
-simplejson
```
Comment (aps-sids) [inline website/requirements.txt]: Again, required by switter.

Comment (funkybob) [inline website/requirements.txt]: Does't switter list its own requirements?  If so, there's no need to list them here.

File: website/website/settings.py (hunk 2 above)
Comment (funkybob) [inline website/website/settings.py]: Again, doesn't switter list its own deps?

Comment (funkybob) [inline website/website/settings.py]: Guess I never got far enough through cleaning up to find out why it was here.

Comment (funkybob) [inline website/website/settings.py]: Their own install docs don't mention its need...

File: website/requirements.txt (hunk 4 above)
Comment (aps-sids) [inline website/requirements.txt]: Switter needs to be updated as well. Also, this - https://github.com/timvideos/streaming-system/blob/master/website/tracker/views.py#L17 - needs to be updated as well.

File: website/website/settings.py (hunk 2 above)
Comment (aps-sids) [inline website/website/settings.py]: yeah, but it's being used here - https://bitbucket.org/mithro/django-switter/src/0c096811b3acf0c90d838b58abccd86fd883c85a/switter/models.py?at=master#cl-2

Comment (funkybob) [inline website/website/settings.py]: Wow, a whole package just so they don't have to type "auto_now=True" and "auto_now_add=True" ?

File: website/website/settings.py (hunk 3 above)
Comment (funkybob) [inline website/website/settings.py]: Ideally you'd have a production settings file import all these settings, and override this.

Leaving this blank avoids problems in development.

**Comment by mithro** [comment]
**If** you get some time, it would be cool for you to take a look at this @micolous

File: website/website/settings.py (hunk 2 above)
Comment (micolous) [inline website/website/settings.py]: Addressing with [this pull request](https://bitbucket.org/mithro/django-switter/pull-requests/2/remove-dependency-on-django_extensions/diff)

File: website/requirements.txt (hunk 4 above)
Comment (micolous) [inline website/requirements.txt]: Addressing with another PR: https://bitbucket.org/mithro/django-switter/pull-requests/2/remove-dependency-on-django_extensions-and/diff

File: website/manage.py (hunk 1 above)
Comment (micolous) [inline website/manage.py]: There is a `__init__.py`, since https://github.com/timvideos/streaming-system/commit/70de77dc76c4de6c7b1d74f9b7af1acbc1c4c95d

**Pull request merged**